src/common/ceph_mutex.h

```diff
@@ -0,0 +1,139 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+
+#pragma once
+
+// What and why
+// ============
+//
+// For general code making use of mutexes, use these ceph:: types.  The main
+// difference is that the ceph::mutex ctor takes a string argument naming
+// the lock for use by lockdep.
+//
+// For legacy Mutex users that passed recursive=true, use ceph::recursive_mutex.
+//
+// For legacy Mutex users that passed lockdep=false, use std::mutex directly.
+
+#ifdef CEPH_DEBUG_MUTEX
+
+// ============================================================================
+// debug (lockdep-capable, various sanity checks and asserts)
+// ============================================================================
+
+#include "common/mutex_debug.h"
+#include "common/condition_variable_debug.h"
+
+namespace ceph {
+  typedef ceph::mutex_debug mutex;
+  typedef ceph::mutex_recursive_debug recursive_mutex;
+  typedef ceph::condition_variable_debug condition_variable;
+}
+
+#else
+
+// ============================================================================
+// fast and minimal
+// ============================================================================
+
+#include <mutex>
+#include <string>
+
+namespace ceph {
+
+  class mutex {
+    std::mutex m;
+
+  public:
+    mutex(const mutex&) = delete;
+    mutex(mutex&) = delete;
+
+    // mutex_debug-style ctor for compat with the CEPH_DEBUG_MUTEX
+    mutex(const std::string &n,
+	  bool bt=false) noexcept {}
```

Comment:
i'd guess it's pretty common for callers to pass in a string literal for the name - think it's worth adding another constructor that takes the first arg as `const char*` to avoid constructing std::strings for those?

Reply:
good idea!  updated.  also make mutex_debug require a name (instead of using the uuid).  Since all users will be moving over from Mutex this seems fine (Mutex requires a name).  And for new users naming these seems like a small burden to continue to bear...